arangod/Aql/LimitExecutor.cpp

```diff
@@ -69,13 +71,37 @@ auto LimitExecutor::calculateUpstreamCall(AqlCall const& clientCall) const -> Aq
                             clientCall.getLimit() < localLimitMinusDownstreamOffset;
 
   if (useSoftLimit) {
+    // fullCount may only be set with a hard limit
+    TRI_ASSERT(!clientCall.needsFullCount());
+
     upstreamCall.softLimit = limit;
     upstreamCall.fullCount = false;
+  } else if (clientCall.needsFullCount() && 0 == clientCall.getLimit() && !_didProduceRows) {
+    // The request is a hard limit of 0 together with fullCount, so we always
+    // need to skip both our offset and our limit, regardless of the client's
+    // offset.
+    // If we ever got a client limit > 0 and thus produced rows, we may not send
+    // a non-zero offset, and thus must avoid this branch.
+    TRI_ASSERT(!useSoftLimit);
+    TRI_ASSERT(clientCall.hasHardLimit());
+
+    upstreamCall.offset = upstreamCall.offset + remainingLimit();
+    upstreamCall.hardLimit = 0;
+    // We need to send fullCount upstream iff this is fullCount-enabled LIMIT
```

Comment:
```suggestion
    // We need to send fullCount upstream if this is fullCount-enabled LIMIT
```

Reply:
@hkernbach I think the original is actually correct.

`iff` is a short-hand for `if and only if` which is correct here.